napari/benchmarks/benchmark_qt_viewer_labels.py

```diff
@@ -83,3 +86,65 @@ def time_fill(self):
     def time_on_mouse_move(self):
         """Time to drag paint on mouse move."""
         self.viewer.window._qt_viewer.canvas._on_mouse_move(self.event)
+
+
+class LabelRenderingSuite:
+    """Benchmarks for rendering the Labels layer."""
+
+    param_names = ["radius", "dtype"]
+    params = ([20, 400, 2000], [np.uint8, np.uint16, np.uint32])
+    if "PR" in os.environ:
+        skip_params = Skiper(lambda x: x[0] >= 100)
+
+    def setup(self, radius, dtype):
+        self.app = QApplication.instance() or QApplication([])
+
+        if radius < 1000:
+            self.data = octahedron(radius=radius, dtype=dtype)
+        else:
+            self.data = np.zeros(
+                (radius // 50, radius * 2, radius * 2), dtype=dtype
+            )
+            for i in range(1, self.data.shape[0] // 2):
+                part = diamond(radius=(i) * 100, dtype=dtype)
+                shift = (self.data.shape[1] - part.shape[0]) // 2
+                self.data[i, shift : -shift - 1, shift : -shift - 1] = part
+                self.data[
+                    -i - 1, shift : -shift - 1, shift : -shift - 1
+                ] = part
+
+        count = np.count_nonzero(self.data)
+        self.data[self.data > 0] = np.random.randint(
+            1, min(2000, np.iinfo(dtype).max), size=count, dtype=dtype
+        )
+        scale = self.data.shape[-1] / np.array(self.data.shape)
+        self.viewer = napari.view_labels(self.data, scale=scale)
+        self.layer = self.viewer.layers[0]
+
+    def teardown(self, *_):
+        self.viewer.window.close()
+
+
+class LabelRenderingSuite2D(LabelRenderingSuite):
+    def setup(self, radius, dtype):
+        super().setup(radius, dtype)
+        self.viewer.dims.ndisplay = 2
+
+    def time_iterate_over_z(self, radius, dtype):
+        """Time to render the layer."""
+        z_size = self.data.shape[0]
+        for i in range(0, z_size, z_size // 20):
+            self.viewer.dims.set_point(0, i)
+            self.app.processEvents()
```

Comment:
```suggestion
class LabelRenderingSuiteInitialLoad(LabelRenderingSuite):
    def setup(self, radius, dtype):
        super().setup(radius, dtype)
        self.viewer.dims.ndisplay = 2

    def time_iterate_over_z(self, radius, dtype):
        """Time to render a 2D slice of the layer repeatedly."""
        z_size = self.data.shape[0]
        for i in range(0, z_size, z_size // 20):
            self.viewer.dims.set_point(0, i)
            self.app.processEvents()
            
    def load_3d(self, radius, dtype):
        """Time to first render of the layer in 3D."""
        self.viewer.dims.ndisplay = 3
        self.app.processEvents()
        self.viewer.dims.ndisplay = 2
```

?

As you've remarked in the PR, the issue is much more pronounced for 3D rendering. That's because 3D volumes present a *lot* more values to convert, and when you are doing it in the CPU, it's sloooow. This addition will measure that slowness.

Reply:
I do not understand this Suggestion. But maybe you forgot to add `time` to the name of the function. (`asv` discover functions with `time` like pytest with `test`)

Reply:
Yes, should be time load 3D